pdudaemon/httpserver.py

```diff
@@ -0,0 +1,107 @@
+#
+#  You should have received a copy of the GNU General Public License
+#  along with this program; if not, write to the Free Software
+#  Foundation, Inc., 51 Franklin Street, Fifth Floor, Boston,
+#  MA 02110-1301, USA.
+
+
+try:
+    from BaseHTTPServer import BaseHTTPRequestHandler, HTTPServer
+    import urlparse
+except ImportError:
+    from http.server import BaseHTTPRequestHandler, HTTPServer
+    import urllib.parse as urlparse
+import logging
+import time
+import sys
+import os
+from pdudaemon.dbhandler import DBHandler
+from pdudaemon.shared import drivername_from_hostname
+log = logging.getLogger(__name__)
+
+class PDUHTTPHandler(BaseHTTPRequestHandler):
+    def _set_headers(self, code):
+        self.send_response(code)
+        self.send_header('Content-type', 'text/plain')
+        self.end_headers()
+
+    def do_GET(self):
+        logging.info("Handling HTTP request from %s: %s" % (self.client_address, self.path))
+        data = urlparse.parse_qs(urlparse.urlparse(self.path).query)
+        res = self.insert_request(data)
+        if res:
+            self._set_headers(200)
+            self.wfile.write("OK - accepted request\n")
+        else:
+            self._set_headers(500)
+            self.wfile.write("Invalid request\n")
+
+    def insert_request(self, data):
+        delay = 10
+        custom_delay = False
+        now = int(time.time())
+        if data.get('delay', None):
+            delay = data.get('delay')[0]
+            custom_delay = True
+        hostname = data.get('hostname', [None])[0]
+        port = data.get('port', [None])[0]
+        request = data.get('command', [None])[0]
+        if not hostname or not port or not request:
+            return False
+        drivername_from_hostname(hostname, self.server.config["pdus"])
+        dbh = self.server.dbh
+        if not (request in ["reboot", "on", "off"]):
+            log.info("Unknown request: %s", request)
+            return False
+        if request == "reboot":
+            log.debug("reboot requested, submitting off/on")
+            dbh.insert_request(hostname, port, "off", now)
+            dbh.insert_request(hostname, port, "on", now + int(delay))
+            return True
+        else:
+            if custom_delay:
+                log.debug("using delay as requested")
+                dbh.insert_request(hostname, port, request, now + delay)
+                return True
+            else:
+                dbh.insert_request(hostname, port, request, now)
+                return True
+
+
+class PDUHTTPListener(object):
+
+    def __init__(self, config):
+        self.config = config
+        settings = config["daemon"]
+        listen_host = settings["hostname"]
+        listen_port = settings["port"]
+        log.debug("PDUHTTPListener __init__")
+        if "purge" in config:
+            log.info("Running a DB purge and exiting.")
+            temp_dbh = DBHandler(settings)
+            temp_dbh.purge()
+            sys.exit(os.EX_OK)
```

Comment:
Seems a bit hacky - why not let `pdudaemon-start` or some other CLI tool call a `purge` method here instead?  Also it doesn't seem like `purge` is actually a config setting, rather a way to execute this code.